taskcluster/ci/tests/kind.yml

```diff
@@ -42,3 +42,18 @@ tasks:
 
           echo "Start the dry run"                                          &&
           make dry-run
+
+  taskgraph-definition:
+      worker-type: b-linux-large
+      worker:
+          docker-image: {in-tree: base}
+          max-run-time: 3600
+      description: "Test the full `translations_taskgraph` to validate the latest changes"
+      run-on-tasks-for: ["github-push", "github-pull-request"]
+      run:
+          command: >-
+              pip3 install -r taskcluster/requirements.txt &&
+              for param in `ls taskcluster/test/params`; do
+                  taskgraph full -p taskcluster/test/params/$param
```

Comment:
Question: Is it possible or even useful to output the taskgraph.json file like you were showing me as an artifact here? Then it would provide an easy mechanism to validate what's being generated manually.

Reply:
It is possible. For validating what is being generated on something like a pull request the `--diff` flag might generate something more useful.

For example when I diff this branch with https://github.com/mozilla/firefox-translations-training/commit/cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d (right before black and lint landed)

`taskgraph full -o taskgraph.diff --diff cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d`

This is the output I get

```
--- full_task_graph@cce2cd7b7c98
+++ full_task_graph@tests-taskgraph-definition
@@ -123,41 +123,44 @@
 fetch-extract-lex
 fetch-fast-align
 fetch-hunspell
 fetch-kenlm
 fetch-marian
 fetch-preprocess
 finetune-student-ru-en
 finetune-teacher-ru-en-1/2
 finetune-teacher-ru-en-2/2
 merge-augmented-ru-en
 merge-corpus-ru-en
 merge-devset-ru-en
 merge-mono-src-ru-en
 merge-mono-trg-ru-en
 merge-translated-ru-en
 quantize-ru-en
 score-ru-en
 split-corpus-ru-en
 split-mono-src-ru-en
 split-mono-trg-ru-en
+tests-black
+tests-lint
 tests-snakemake-dry-run
+tests-taskgraph-definition
 toolchain-browsermt-marian
 toolchain-cuda-toolkit
 toolchain-extract-lex
 toolchain-fast-align
 toolchain-hunspell
 toolchain-kenlm
 toolchain-marian
 toolchain-preprocess
 train-backwards-ru-en
 train-student-ru-en
 train-teacher-ru-en-1/2
 train-teacher-ru-en-2/2
 train-vocab-ru-en
 translate-corpus-ru-en-1/10
 translate-corpus-ru-en-10/10
 translate-corpus-ru-en-2/10
 translate-corpus-ru-en-3/10
 translate-corpus-ru-en-4/10
 translate-corpus-ru-en-5/10
 translate-corpus-ru-en-6/10
Files full_task_graph@cce2cd7b7c98 and full_task_graph@tests-taskgraph-definition are identical
```

So we can see what tasks you are adding and removing (it also supports parameters if you want to see the tasks for specific events.)

`taskgraph target-graph -p parameters.yml -o taskgraph.diff --diff cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d`

There's not really a good way to surface the artifact to the `PR` at the moment.

Reply:
It is possible. For validating what is being generated on something like a pull request the `--diff` flag might generate something more useful.

For example when I diff this branch with https://github.com/mozilla/firefox-translations-training/commit/cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d (right before black and lint landed)

`taskgraph full -o taskgraph.diff --diff cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d`

This is the output I get

```
--- full_task_graph@cce2cd7b7c98
+++ full_task_graph@tests-taskgraph-definition
@@ -123,41 +123,44 @@
 fetch-extract-lex
 fetch-fast-align
 fetch-hunspell
 fetch-kenlm
 fetch-marian
 fetch-preprocess
 finetune-student-ru-en
 finetune-teacher-ru-en-1/2
 finetune-teacher-ru-en-2/2
 merge-augmented-ru-en
 merge-corpus-ru-en
 merge-devset-ru-en
 merge-mono-src-ru-en
 merge-mono-trg-ru-en
 merge-translated-ru-en
 quantize-ru-en
 score-ru-en
 split-corpus-ru-en
 split-mono-src-ru-en
 split-mono-trg-ru-en
+tests-black
+tests-lint
 tests-snakemake-dry-run
+tests-taskgraph-definition
 toolchain-browsermt-marian
 toolchain-cuda-toolkit
 toolchain-extract-lex
 toolchain-fast-align
 toolchain-hunspell
 toolchain-kenlm
 toolchain-marian
 toolchain-preprocess
 train-backwards-ru-en
 train-student-ru-en
 train-teacher-ru-en-1/2
 train-teacher-ru-en-2/2
 train-vocab-ru-en
 translate-corpus-ru-en-1/10
 translate-corpus-ru-en-10/10
 translate-corpus-ru-en-2/10
 translate-corpus-ru-en-3/10
 translate-corpus-ru-en-4/10
 translate-corpus-ru-en-5/10
 translate-corpus-ru-en-6/10
Files full_task_graph@cce2cd7b7c98 and full_task_graph@tests-taskgraph-definition are identical
```

So we can see what tasks we are adding and removing (it also supports parameters if you want to see the tasks for specific events.)

`taskgraph target-graph -p parameters.yml -o taskgraph.diff --diff cce2cd7b7c98bee6512a8ccfd24b98e1ac51ff6d`

There's not really a good way to surface the artifact to the `PR` at the moment.